consensus/ibft/ibft.go

```diff
@@ -393,7 +393,7 @@ func (i *Ibft) buildBlock(snap *Snapshot, parent *types.Header) (*types.Block, e
 			i.txpool.DecreaseAccountNonce(txn)
 		} else {
 			if err := transition.Write(txn); err != nil {
-				if err.IsRecoverable {
+				if err.(*state.TransitionApplicationError).IsRecoverable {
```

Comment:
Just a comment. It looks to be ok now. But you might want to add safe-check.

```go
if appErr, ok := err.(*state.TransitionApplicationError); ok && appErr.IsRecoverable {
```

Reply:
Good call 👍 

It's always better to add that additional type assertion check layer 💯 

Added:
f9bfe91